components/gateway/src/main/kotlin/net/corda/p2p/gateway/messaging/http/TrustStoresMap.kt

```diff
@@ -55,11 +73,11 @@ internal class TrustStoresMap(
 
     fun getTrustStore(sourceX500Name: MemberX500Name, destinationGroupId: String) =
         trustRootsPerHoldingIdentity[TruststoreKey(sourceX500Name, destinationGroupId)]
-            ?.trustStore
+            ?.cachedKeyStore()
             ?: throw IllegalArgumentException("Unknown trust store for source X500 name ($sourceX500Name) " +
                     "and group ID ($destinationGroupId)")
 
-    fun getTrustStores() = trustRootsPerHoldingIdentity.values.mapNotNull { it.trustStore }
+    fun getTrustStores() = trustRootsPerHoldingIdentity.values.mapNotNull { it.cachedKeyStore() }
```

Comment:
if you convert `cachedKeyStore` to return a non-nullable type, this could also be simplified to:
```suggestion
    fun getTrustStores() = trustRootsPerHoldingIdentity.values.map { it.cachedKeyStore() }
```